### PR TITLE
libc: Move queue.h from include to include/nuttx

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
to avoid the conflict with libuv's queue.h.
Replace the patch https://github.com/apache/incubator-nuttx-apps/pull/1321
need merge with https://github.com/apache/incubator-nuttx-apps/pull/1323 together

## Impact
Code refactor

## Testing
Pass CI

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
Please ignore the mix case warning from arch/arm/src/phy62xx.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Comment by pkarashchenko** [comment]
Seems like some files still need to be updated
```
In file included from wdog/wd_gettime.c:30:
Error: /github/workspace/sources/nuttx/sched/wdog/wdog.h:75:8: error: unknown type name 'sq_queue_t'
   75 | extern sq_queue_t g_wdactivelist;
      |        ^~~~~~~~~~
```
I will try to run the build locally and feedback

File: include/nuttx/wdog.h [hunk 1]
```diff
@@ -29,7 +29,6 @@
 
 #include <nuttx/clock.h>
 #include <stdint.h>
-#include <queue.h>
```
Comment (pkarashchenko) [inline include/nuttx/wdog.h:32]: Seems now we need to add `#include <nuttx/queue.h>`  to `sched/wdog/wdog.h` as it was implicitly included till now

Comment (xiaoxiang781216) [reply include/nuttx/wdog.h:32]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**